File: 119. Pascal's Triangle II/Solution.py
# LeetCode #119: Pascal's Triangle II

'''
Given an index k, return the kth row of the Pascal's triangle.

For example, given k = 3,
Return [1,3,3,1].

Note:
Could you optimize your algorithm to use only O(k) extra space?
'''

class Solution:
    # @param {integer} rowIndex
    # @return {integer[]}
    def getRow(self, rowIndex):
        res = []
        if rowIndex < 0:
            return res
            
        res.append(1)
        for i in range(0, rowIndex):
            for j in range(len(res) - 1, 0, -1):
                res[j] += res[j-1]
            res.append(1)
            
        return res
        

File: 119. Pascal's Triangle II/Solution.java
// LeetCode #119: Pascal's Triangle II

/*
Given an index k, return the kth row of the Pascal's triangle.

For example, given k = 3,
Return [1,3,3,1].

Note:
Could you optimize your algorithm to use only O(k) extra space?
*/

public class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> res = new ArrayList<Integer>();
        if (rowIndex < 0)
            return res;
            
        res.add(1);
        for(int i = 0; i < rowIndex; i++) {
            for(int j = res.size() - 1; j > 0; j--) {
                res.set(j, res.get(j) + res.get(j-1));
            }
            res.add(1);
        }
        return res;
    }
}